libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioAdapter.cs

```diff
@@ -41,18 +41,19 @@ public class TwilioAdapter : BotAdapter, IBotFrameworkHttpAdapter
         /// </remarks>
         /// <param name="logger">The ILogger implementation this adapter should use.</param>
         public TwilioAdapter(IConfiguration configuration, ILogger logger = null)
-            : this(new TwilioClientWrapper(new TwilioAdapterOptions(configuration["TwilioNumber"], configuration["TwilioAccountSid"], configuration["TwilioAuthToken"], new Uri(configuration["TwilioValidationUrl"]))), logger)
+            : this(new TwilioAdapterOptions(configuration["TwilioNumber"], configuration["TwilioAccountSid"], configuration["TwilioAuthToken"], new Uri(configuration["TwilioValidationUrl"])), logger)
         {
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TwilioAdapter"/> class.
         /// </summary>
-        /// <param name="twilioClient">The Twilio client to connect to.</param>
+        /// <param name="options">Options for the Twilio Adapter.</param>
         /// <param name="logger">The ILogger implementation this adapter should use.</param>
-        public TwilioAdapter(TwilioClientWrapper twilioClient, ILogger logger = null)
+        /// <param name="twilioClient">Client to use for the Twilio API.</param>
+        public TwilioAdapter(TwilioAdapterOptions options, ILogger logger = null, TwilioClientWrapper twilioClient = null)
```

Comment:
The constructor doesn't look very clean, options and twilioClient seem redundant in some scenarios. 
For example, if I provide the twiluiClient, then I shouldn't need options. 

Would it be possible to have just one of them? if not, it may be clearer if we have two constructors, one with options and no client and one with client and no options.